third_party/txt/src/minikin/FontCollection.cpp

```diff
@@ -509,7 +509,9 @@ void FontCollection::itemize(const uint16_t* string,
     if (!shouldContinueRun) {
       const std::shared_ptr<FontFamily>& family = getFamilyForChar(
           ch, isVariationSelector(nextCh) ? nextCh : 0, langListId, variant);
-      if (utf16Pos == 0 || family.get() != lastFamily) {
+
+      if (utf16Pos == 0 || family.get() != lastFamily ||
+          family.get()->getLastMatchedCodePoint() != ch) {
```

Comment:
This algorithm tries to reuse the already selected font as much as possible.
I think that using only one codepoint cached causes too much execution.

Reply:
You're right. Maybe I should use map for this.